api/v1alpha4/machinedeployment_types.go

```diff
@@ -69,8 +69,7 @@ type MachineDeploymentSpec struct {
 
 	// The deployment strategy to use to replace existing machines with
 	// new ones.
-	// +optional
-	Strategy *MachineDeploymentStrategy `json:"strategy,omitempty"`
+	Strategy *MachineDeploymentStrategy `json:"strategy"`
```

Comment:
Is it right that we are still using a pointer if the intent is to make this field required?
Also, if a user today doesn't add `strategy` to their yaml, they gets the default strategy. Are we preserving this behavior or are we facing the user to add `strategy. {}` to get defaults?

Reply:
>Is it right that we are still using a pointer if the intent is to make this field required?

It still needs to be a pointer to not break clients.

>Also, if a user today doesn't add strategy to their yaml, they gets the default strategy. Are we preserving this behavior or are we facing the user to add strategy. {} to get defaults?

Yeh The default still applies, no changes there.